### ISSUE TITLE
Package Generator: Fix and Verify the process to create a new dictionary

### ISSUE THREAD

**Issue by Jason3S** [streetsidesoftware/cspell-dicts#75]
# Issues:

## Package Generation:
The build process must depend upon the type of resulting dictionary.
- If it is a `*.txt.gz` file, then we want to build it each time.
- If it is a `*.trie.gz` file, then we only want to build it by hand because it can take a very long time.
- `test` script depends upon the type of source, for a hunspell file, we need to use the hunspell reader, for a text file, a standard `head` will work.
   **Example Test Text**
  ```
  "test": "head -n 100 \"go.txt\" | cspell -v -c ./cspell-ext.json --local=* --languageId=go stdin",
  ```
  **Example Test Hunspell**
  ```
  "test": "hunspell-reader words -n 1000 \"French.dic\" | cspell -v -c ./cspell-ext.json --local=fr --languageId=* stdin",
  ```
  
## Building / testing:
- The build process is broken for running in the package directory `npm run build` will fail from the package directory if `cspell`, `cspell-tools`, and `hunspell-reader` are not installed globally.

## Documentation
- Documentation needs to be updated to reflect the new process.

**Comment by d2s** [comment]
Yeah, noticed the same problem with missing `cspell-tools` when doing some improvements to the companies dictionary https://github.com/streetsidesoftware/cspell-dicts/pull/106 earlier today.

It might be a good early step to add pointer to this issue from the `README.md`, together with a mention that `cspell`, `cspell-tools`, and `hunspell-reader` are required for building packages.
